cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/ResourceDistributionGoal.java

```diff
@@ -321,22 +408,210 @@ protected boolean rebalanceByMovingLoadIn(Broker broker,
     return true;
   }
 
-  protected boolean rebalanceByMovingLoadOut(Broker broker,
+  /**
+   * Get the sorted replicas in the given broker whose (1) topic is not an excluded topic AND (2) do not violate the given load
+   * limit in ascending or descending order. Load limit requires the replica load to be (1) above the given limit in
+   * descending order, (2) below the given limit in ascending order.
+   *
+   * @param broker Broker whose replicas will be considered.
+   * @param excludedTopics Excluded topics for which the replicas will be remove from the returned candidate replicas.
+   * @param loadLimit Load limit determining the lower cutoff in descending order, upper cutoff in ascending order.
+   * @param isDescending True if sort requested in descending order, false otherwise.
+   * @return Sorted replicas in the given broker whose (1) topic is not an excluded topic AND (2) do not violate the
+   * given load limit in ascending or descending order.
+   */
+  private SortedSet<Replica> sortedCandidateReplicas(Broker broker,
+                                                     Set<String> excludedTopics,
+                                                     double loadLimit,
+                                                     boolean isDescending) {
+    SortedSet<Replica> candidateReplicas = new TreeSet<>((r1, r2) -> {
+      int result = isDescending
+                   ? Double.compare(r2.load().expectedUtilizationFor(resource()), r1.load().expectedUtilizationFor(resource()))
+                   : Double.compare(r1.load().expectedUtilizationFor(resource()), r2.load().expectedUtilizationFor(resource()));
+      return result == 0 ? r1.topicPartition().toString().compareTo(r2.topicPartition().toString()) : result;
+    });
+
+    // If the resource is NW_OUT, candidate replicas consider only the leaders -- i.e. only the leaders have NW_OUT load,
+    // otherwise all replicas on broker are considered.
+    Set<Replica> coveredReplicas = resource() == Resource.NW_OUT ? broker.leaderReplicas() : broker.replicas();
+
+    // The given load limit determines the lower cutoff in descending order, upper cutoff in ascending order.
+    if (isDescending) {
+      candidateReplicas.addAll(coveredReplicas.stream()
+                                              .filter(l -> (!excludedTopics.contains(l.topicPartition().topic()))
```

Comment:
We are using excludedTopics directly instead of shouldExclude(). Is this expected?